src/tools/annotations.R

```diff
@@ -239,8 +239,14 @@ annotate_data <- function(expression_set, chip_id, selections) {
     ))
   }
   
-  # TODO : Maybe print out the number of NAs in the annotations for each
-  # type of annotation?
+  # Print out the number of NAs in the annotations for each type of annotation
+  notMap = matrix(0, nrow = 2, ncol = dim(annotations)[2],
+                  dimnames = list(c("NA entries","%"), colnames(annotations)))
+  for (i in colnames(annotations)) {
+    notMap[1,i] = sum(isNA(annotations[,i]))
+    notMap[2,i] = round(notMap[1,i]/dim(annotations)[1]*1e2, digits = 2)
+  }
+  log_info(paste("Missing annotations:", get.print.str(notMap), sep = "\n"))
```

Comment:
Logging *should* not be divided into multiple lines. Maybe pipe the `paste`d final string to a `gsub` call to replace newlines with spaces or some other character?

Reply:
(see below)